legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/from/mapping/MappingParseTreeWalker.java

```diff
@@ -156,35 +176,94 @@ else if (mappingElement instanceof AssociationMapping)
         return mapping;
     }
 
-    private MappingTest visitMappingTest(MappingParserGrammar.TestContext ctx, Mapping mapping)
+    private MappingTestSuite visitMappingTestSuite(MappingParserGrammar.MappingTestSuiteContext ctx, Mapping mapping)
+    {
+        MappingTestSuite mappingTestSuite = new MappingTestSuite();
+        mappingTestSuite.sourceInformation = this.walkerSourceInformation.getSourceInformation(ctx);
+        mappingTestSuite.id = PureGrammarParserUtility.fromIdentifier(ctx.identifier());
+
+        List<MappingParserGrammar.MappingTestDataContext> validatedDataCtx = PureGrammarParserUtility.validateRequiredListField(ctx.mappingTestData(), "data", mappingTestSuite.sourceInformation);
+        mappingTestSuite.storeTestDatas = ListIterate.collect(validatedDataCtx, this::visitMappingStoreTestData);
+        List<MappingParserGrammar.MappingTestContext> validatedTestCtx = PureGrammarParserUtility.validateRequiredListField(ctx.mappingTest(), "tests", mappingTestSuite.sourceInformation);
+        mappingTestSuite.tests = ListIterate.collect(validatedTestCtx, atomicTest -> this.visitAtomicTest(atomicTest, mapping));
+
+        return mappingTestSuite;
+    }
+
+    private StoreTestData visitMappingStoreTestData(MappingParserGrammar.MappingTestDataContext ctx)
+    {
+        StoreTestData testData = new StoreTestData();
+        testData.data = HelperEmbeddedDataGrammarParser.parseEmbeddedData(ctx.embeddedData(), this.walkerSourceInformation, this.parserContext.getPureGrammarParserExtensions());
+        this.createDummyBindingIfRequired(testData.data);
+        testData.store = ctx.qualifiedName().packagePath() == null && testData.data instanceof ModelStoreData ?
+                "ModelStore" :
+                PureGrammarParserUtility.fromQualifiedName(ctx.qualifiedName().packagePath().identifier(),ctx.qualifiedName().identifier());     //build store
+        return testData;
+    }
+
+    private void createDummyBindingIfRequired(EmbeddedData data)
+    {
+        if (data instanceof ModelStoreData)
```

Comment:
this is the only place we create the binding, we dont create in model store data, there we just create a pointer, because there is no elementConsumer which can help us put binding into the process

Reply:
yes so I meant lets extract binding from data and consume here why create binding object again